lib/cipher.rb

```diff
@@ -1,9 +1,66 @@
 require_relative 'shift_generator'
+require_relative './modules/create_defaults'
 
 class Cipher
-  attr_reader :shift_key
-  
+  include Generable
+
+  attr_reader :shift_generator,
+              :character_set,
+              :default_key,
+              :default_date
+
   def initialize
-    @shift_key = ShiftGenerator.new
+    @default_key = generate_random_key
+    @default_date = create_date
+    @shift_generator = ShiftGenerator.new
+    @character_set = create_character_set
+  end
+
+  def chop_message(message)
+    message.downcase.split("").each_slice(4).to_a
+  end
+
+  def encrypt_letter(letter, shift)
+    encrypted_set = Hash[@character_set.zip(@character_set.rotate(shift))]
+    return letter if !encrypted_set.keys.include?(letter)
+    encrypted_set.fetch(letter)
+  end
+
+  def encrypt_chop(chop, shift_key)
+    chop.map.with_index do |letter, index|
+      encrypt_letter(letter, shift_key[index])
+    end
+  end
+
+  def encrypt_message(message, shift)
+    chop_message(message).map do |chunk|
```

Comment:
this method puts the above two together. the reasoning for this is to keep method lengths shorter and in granular chunks that makes sense to me, might be a tad redundant though, will consider refactoring